src/sparseml/pytorch/torchvision/train.py

```diff
@@ -0,0 +1,949 @@
+# Copyright (c) 2021 - present / Neuralmagic, Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import datetime
+import math
+import os
+import time
+import warnings
+
+import torch
+import torch.utils.data
+import torchvision
+from torch import nn
+from torch.utils.data.dataloader import default_collate
+from torchvision.transforms.functional import InterpolationMode
+
+from sparseml.pytorch.models.registry import ModelRegistry
+from sparseml.pytorch.optim import ScheduledModifierManager
+from sparseml.pytorch.torchvision import presets, transforms, utils
+from sparseml.pytorch.torchvision.sampler import RASampler
+from sparseml.pytorch.utils.helpers import (
+    download_framework_model_by_recipe_type,
+    torch_distributed_zero_first,
+)
+from sparsezoo import Model
+
+
+def train_one_epoch(
+    model,
+    criterion,
+    optimizer,
+    data_loader,
+    device,
+    epoch,
+    args,
```

Comment:
nit: This is okay for now, but imo passing around this name space object makes debugging difficult down the line

Reply:
Agree, a number of the functions in this file can be refactored